src/Sulu/Bundle/AdminBundle/Metadata/GroupProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Metadata;
+
+use Sulu\Article\Domain\Model\ArticleInterface;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FormGroup;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\TypedFormMetadata;
+
+final readonly class GroupProvider implements GroupProviderInterface
+{
+    public function __construct(
+        private MetadataProviderRegistry $metadataProviderRegistry,
+    ) {
+    }
+
+    public function getGroups(): array
+    {
+        /** @var TypedFormMetadata $metadata */
+        $metadata = $this->metadataProviderRegistry->getMetadataProvider('form')
+            ->getMetadata(ArticleInterface::TEMPLATE_TYPE, '', []);
+
+        $groups = [];
+
+        foreach ($metadata->getForms() as $articleForm) {
+            $group = $articleForm->getGroup() ?: 'default';
+
+            if (!\array_key_exists($group, $groups)) {
+                $groups[$group] = new FormGroup($group, $this->getGroupTitle($group));
+            }
+            $groups[$group] = $groups[$group]->withTemplate($articleForm->getKey());
+        }
+
+        return $groups;
+    }
+
+    private function getGroupTitle(string $groupIdentifier): string
+    {
+        return \ucfirst($groupIdentifier);
```

Comment:
I think it would make sense to use a translation here, but that can be improved in another PR